what4/src/What4/Expr/Builder.hs

```diff
@@ -5263,10 +5265,18 @@ instance IsExprBuilder (ExprBuilder t st fs) where
     | x == y = return $! truePred sym
     | otherwise = (floatIEEELogicBinOp FloatEq) sym x y
   floatNe sym x y = notPred sym =<< floatEq sym x y
-  floatFpEq = floatIEEELogicBinOp FloatFpEq
-  floatFpNe = floatIEEELogicBinOp FloatFpNe
-  floatLe = floatIEEELogicBinOp FloatLe
-  floatLt = floatIEEELogicBinOp FloatLt
+  floatFpEq sym x y
+    | x == y = notPred sym =<< floatIsNaN sym x
+    | otherwise = floatIEEELogicBinOp FloatFpEq sym x y
+  floatFpNe sym x y
+    | x == y = return $ falsePred sym
+    | otherwise = floatIEEELogicBinOp FloatFpNe sym x y
```

Comment:
Is the reason for the new logic here that `==` is faster than `floatIEEELogicBinOp`?

Reply:
`floatIEEELogicBinOp` doesn't evaluate the semantics of the operation at all... it literally just applies the given syntax former.  This just captures some common special cases.

Reply:
@robdockins Gotcha. Thanks!